libs/galley-types/src/Galley/Types.hs

```diff
@@ -1108,3 +1129,18 @@ instance FromJSON ConversationCode where
         ConversationCode <$> o .: "key"
             <*> o .:  "code"
             <*> o .:? "uri"
+
+-- TODO generic instance?
```

Comment:
it's [here](https://hackage.haskell.org/package/aeson-1.4.6.0/docs/Data-Aeson-TH.html#v:deriveJSON), but @arianvp just convinced me we shouldn't do it: better write manual instances and round-trip tests, then we'll get more feedback from the type checker on what to keep in mind when we change the type.  also it already works.  (just need to add an `Arbitrary` instance [here](https://github.com/wireapp/wire-server/blob/1409d5882e6d1f12728e14fff52961d79ae95526/libs/brig-types/src/Brig/Types/Test/Arbitrary.hs) and a line for the test [here](https://github.com/wireapp/wire-server/blob/1409d5882e6d1f12728e14fff52961d79ae95526/libs/brig-types/test/unit/Test/Brig/Types/Common.hs#L59).)

([note that galley-types is tested in brig-types, for complicated reasons](https://github.com/wireapp/wire-server/blob/1409d5882e6d1f12728e14fff52961d79ae95526/libs/brig-types/test/unit/Test/Brig/Types/Common.hs#L7-L8))

Reply:
I agree with sticking to manual instances for now. Otherwise it's too easy to break the JSON structure without noticing.